plugin/pkg/auth/authorizer/node/node_authorizer.go

```diff
@@ -0,0 +1,169 @@
+		// reject requests that don't fit within the broad authorization rules
+		return false, "", nil
+	}
+
+	if len(nodeName) == 0 {
+		// allow requests from unidentifiable nodes if not in strict mode
+		return true, "", nil
+	}
+
+	// subdivide access to specific resources
+
+	if attrs.GetAPIGroup() != api.GroupName {
+		// the resources we care about subdividing happen to be in the core API group.
+		// access to resources in other API groups is not subdivided.
+		return true, "", nil
+	}
+
+	switch attrs.GetResource() {
+	case "configmaps":
+		if attrs.GetVerb() != "get" || len(attrs.GetName()) == 0 || len(attrs.GetNamespace()) == 0 {
+			glog.V(2).Infof("NODE DENY: %#v", attrs)
+			return false, "can only get individual configmaps", nil
+		}
+		return r.hasPathFrom(nodeName, ConfigMapNode, attrs.GetNamespace(), attrs.GetName()), "", nil
+
+	case "secrets":
+		if attrs.GetVerb() != "get" || len(attrs.GetName()) == 0 || len(attrs.GetNamespace()) == 0 {
+			glog.V(2).Infof("NODE DENY: %#v", attrs)
+			return false, "can only get individual secrets", nil
+		}
+		return r.hasPathFrom(nodeName, SecretNode, attrs.GetNamespace(), attrs.GetName()), "", nil
+
+	case "persistentvolumeclaims":
+		if attrs.GetVerb() != "get" || len(attrs.GetName()) == 0 || len(attrs.GetNamespace()) == 0 {
+			glog.V(2).Infof("NODE DENY: %#v", attrs)
+			return false, "can only get individual persistentvolumeclaims", nil
+		}
+		return r.hasPathFrom(nodeName, PVCNode, attrs.GetNamespace(), attrs.GetName()), "", nil
+
+	case "persistentvolumes":
+		if attrs.GetVerb() != "get" || len(attrs.GetName()) == 0 {
+			glog.V(2).Infof("NODE DENY: %#v", attrs)
+			return false, "can only get individual persistentvolumes", nil
+		}
+		return r.hasPathFrom(nodeName, PVNode, "", attrs.GetName()), "", nil
+
+	default:
+		// Access to other resources is not subdivided
+		return true, "", nil
+	}
+}
+
+// hasPathFrom returns true if there is a directed path from the specified type/namespace/name to the specified Node
+func (r *NodeAuthorizer) hasPathFrom(nodeName string, startingType nodeType, startingNamespace, startingName string) bool {
+	r.graph.lock.RLock()
+	defer r.graph.lock.RUnlock()
+
+	nodeNode, exists := r.graph.getNode(NodeNode, "", nodeName)
+	if !exists {
+		glog.V(2).Infof("NODE DENY: unknown node %s asked for %s %s/%s", nodeName, nodeTypes[startingType], startingNamespace, startingName)
+		return false
+	}
+
+	startingNode, exists := r.graph.getNode(startingType, startingNamespace, startingName)
+	if !exists {
+		glog.V(2).Infof("NODE DENY: node %s asked for unknown %s %s/%s", nodeName, nodeTypes[startingType], startingNamespace, startingName)
+		return false
+	}
+
+	found := false
+	traversal := &traverse.VisitingDepthFirst{
+		EdgeFilter: func(edge graph.Edge) bool {
+			if nodeEdge, ok := edge.(*NodeEdge); ok {
+				if nodeEdge.NodeID() == nodeNode.ID() {
+					// We found an edge leading to the node we want
+					found = true
+					return true
```

Comment:
If you found a matching edge, why are you continuing traversal?  The return doesn't do what I think it does?